### ISSUE TITLE
Build fails with rosbash errors from build_env.sh with CATKIN_SHELL=bash

### ISSUE THREAD

**Issue by dornhege** [catkin/catkin_tools#270]
When I try to build a workspace I get the following errors for each package:

    ...build_env.sh: 6: /opt/ros/hydro/share/rosbash/rosbash: function: not found
    ...build/<pkg>/build_env.sh: 7: /opt/ros/hydro/share/rosbash/rosbash: [[: not found
    ...build/<pkg>/build_env.sh: 7: /opt/ros/hydro/share/rosbash/rosbash: Linux: not found
    sed: unrecognized option '--extend'
    ...

These lines in /opt/ros/hydro/share/rosbash/rosbash just contain some standard `bash` features, like `[[`.

The problem is caused by build_env.sh starting with `#!/usr/bin/env sh`, which on my system is a `dash`, not `bash` and somehow rosbash being "active".

I can fix that by editing `verbs/catkin_build/common.py` and changing that to `#!/usr/bin/env bash`. However, I'm not sure if that is the correct fix, especially as that should prevent anyone from building.

I'm not aware of any unusual system setup that I have, but it might be possible. What would cause such a behavior and is there a proper fix or is this it?

**Comment by jbohren** [comment]
> I'm not aware of any unusual system setup that I have, but it might be possible. What would cause such a behavior and is there a proper fix or is this it?

@dornhege It's strange that you're seeing this error, but it appears that `build_env.sh` is failing when it calls `rosbash` through the `rosbash` catkin env-hooks. Can you give a little more information about your system/install/environment? Also can you try running the unit tests? You can see instructions to run the tests here: https://github.com/catkin/catkin_tools/blob/master/tests/README.md

Also, do you have the same problem with the pre-0.4.0 release? It does away with `build_env.sh` entirely. See the prerelease PR here: https://github.com/catkin/catkin_tools/pull/249

```bash
git clone git@github.com:catkin/catkin_tools.git -b pre-0.4.0-executor
```

**Comment by dornhege** [comment]
catkin env-hooks was a good tip! I don't know exactly, what calls what to determine to root cause or fix of this, but I figured out that I do set `CATKIN_SHELL=bash` (for something else). If I unset this, `catkin build` works fine.

I was running 0.3.1 on my system, which should be the same in the current master. I get the same errors for the source version. Running the tests gives me similar errors (besides some about flake8 missing, which I assume is unrelated). pre-0.4.0-executor does not work for me as it is missing `osrf_pycommon.process_utils`. I found `osrf-common` and `osrf-common-hydro` and installed that, but that seems not enough.

As I can trigger this by setting/unsetting `CATKIN_SHELL`, the obvious question would be: Is this supposed to work with `CATKIN_SHELL=bash`?

**Comment by jbohren** [comment]
> I was running 0.3.1 on my system, which should be the same in the current master. I get the same errors for the source version. Running the tests gives me similar errors (besides some about flake8 missing, which I assume is unrelated). pre-0.4.0-executor does not work for me as it is missing `osrf_pycommon.process_utils`. I found `osrf-common` and `osrf-common-hydro` and installed that, but that seems not enough.

Ah sorry about that, you'd also need to install this from source to try 0.4.0: https://github.com/osrf/osrf_pycommon

> As I can trigger this by setting/unsetting `CATKIN_SHELL`, the obvious question would be: Is this supposed to work with `CATKIN_SHELL=bash`?

It should definitely work without having to set that. What OS/version are you on?

**Comment by dornhege** [comment]
0.4.0 seems to work. Is there any way to check, what version I'm running to not accidentally use my system install? I believe the last output rows with Warnings: None... are new, correct?

To be clear: If I set `CATKIN_SHELL=bash` catkin tools (from master) does NOT work. If I unset this, it works like it should. So me setting `CATKIN_SHELL=bash` triggers this bug. I have to set that for something unrelated, not for catkin build. I'm running on Precise/12.04 with hydro and catkin tools 0.3.1.

**Comment by jbohren** [comment]
> 0.4.0 seems to work. Is there any way to check, what version I'm running to not accidentally use my system install? I believe the last output rows with Warnings: None... are new, correct?

Correct. 

> To be clear: If I set CATKIN_SHELL=bash catkin tools (from master) does NOT work. If I unset this, it works like it should. So me setting CATKIN_SHELL=bash triggers this bug. I have to set that for something unrelated, not for catkin build. I'm running on Precise/12.04 with hydro and catkin tools 0.3.1.

Oh I see, I misunderstood that. Does it have a similar effect on pre-0.4.0?

**Comment by dornhege** [comment]
pre-0.4.0 works fine with `CATKIN_SHELL=bash`.

**Comment by jbohren** [comment]
This is fixed after merging #249.
